server/sandbox_stop_linux.go

```diff
@@ -94,7 +93,6 @@ func (s *Server) stopPodSandbox(ctx context.Context, sb *sandbox.Sandbox) error
 
 	log.Infof(ctx, "Stopped pod sandbox: %s", sb.ID())
 	sb.SetStopped(ctx, true)
-	s.generateCRIEvent(ctx, sb.InfraContainer(), types.ContainerEventType_CONTAINER_STOPPED_EVENT)
```

Comment:
Does the kubelet expect a stop for pod infra container or does it need a separate pod stop event?

Reply:
CRI events are generated per container, which also includes infra container. If for some reason infra container runs to completion the `ContainerEventType_CONTAINER_STOPPED_EVENT` will be generated by crio in [handleExit()](https://github.com/cri-o/cri-o/blob/812bc5d664d4ec21c01ff172bce1a2d83f88d5c8/server/server.go#L899) function. This event will be [received and forwarded](https://github.com/kubernetes/kubernetes/blob/8d25cc319391450d9e8b27db0ff8e288da926340/pkg/kubelet/pleg/evented.go#L274) to the syncLoop in Kubelet in the form of `PodLifeCycle` event. 


This PR is fixing the issue where we were incorrectly generating `ContainerEventType_CONTAINER_STOPPED_EVENT` while intentionally stopping the infra container rather than generating it when that infra container runs to completion (for which the relevant code in `handleExit()` already existed as mentioned above)

In retrospect, I feel we should have named `ContainerEventType_CONTAINER_STOPPED_EVENT` as `ContainerEventType_CONTAINER_EXITED_EVENT` because it is generating some confusion with action that is stopping the container (e.g. https://github.com/cri-o/cri-o/pull/6531#issuecomment-1409365765)

Reply:
@haircommander do you have a reproducer for this scenario?

Reply:
if you create a pod that doesn't share a pod level pid namespace (`shareProcessNamespace: false`) then the infra container will be spoofed

Reply:
Thanks @haircommander , I added a comment describing this scenario in the code.

Reply:
@sairameshv can you only call this
`if podInfraContainer.Spoofed()`? 

otherwise, there will be two stopped events when the infra container is not dropped (one from handleExit, and one from here)

Reply:
Yeah @haircommander , Updated.